README.md
<h1 align="center">Python MiniGames (CUI) 👨🏻‍💻 </h1>

A minimal fun CUI based mini games written in python. Arcade includes 5 mini games at present(30/12/21). 
All of the games uses random python package, these games are not intended to be super cooool, they are just made for creative purpose. All of the 5 games are super simple and fun to play with friends.

<hr style="border:10px solid grey"> </hr>
<br><h2 align="center">1. Guess the number ⬆️ or ⬇️</h1>
<p> <b>Description:</b>
  rules are simple, you can choose from 1 to 4 players, you are asked to choose a range in which the lucky number is choosen. You're task is simple guess the lucky    number.
    each player gets 1 chance in a round, after 2 cycles the number is changed. Player who guess's the lucky number wins.</p>

## Preview: ##
<p align="center">
<img src="preview img/numguess.png" align="center" height="700px">
</p><br>
<hr style="border:2px solid gray"> </hr>
<br><h2 align="center">2. Casino Simulator 🤞🎲🎰</h1>
<p> <b>Description:</b>
  There is no limit for no.of players. You are asked to choose a range in which 15 random number are selected and displayed, $100 per player, you're task is to guess the one lucky number from the list and place bet on that number, u can place bet on any no.of number within 15 by any denomination withtin you're $100.
  player with the greater amount at the end of the game wins.
</p>

## Preview: ##
<p align="center">
<img src="preview img/casio.png" align="center" height="700px">
</p><br>
<hr style="border:2px solid gray"> </hr>
<br><h2 align="center">3. Lottery Simulator 🤑💸💰🎟️</h1>
<p> <b>Description:</b>
  Max of 4 player can play this game. In this game there is nothing to do much, you enter no.of player and wait till the computer simulates a lottery winner.
  after a couple of iterations, permutaions and combinations winner is displayed with the no.of points he/she gained. 
</p>

## Preview: ##
<p align="center">
<img src="preview img/lotterysim.png" align="center" >
</p><br>
<hr style="border:2px solid gray"> </hr>
<br><h2 align="center">4. Binary kriket 🏏</h1>
<p> <b>Description:</b>
  a single player game. we all played the nostalsgic hand cricket in our childhood, this game imitates the same environment. you are asked to choose toss, if you win the toss you're given an option to choose b/w bat or bowl or vice versa applies for the computer. you cast you're runs until on of you(computer/you) gets bowled. One with majority runs at the end of the game wins.
</p>

## Preview: ##
<p align="center">
<img src="preview img/binkriket.png" align="center" height="700px" >
</p><br>
<hr style="border:2px solid gray"> </hr>
<br><h2 align="center">5. Stock-market Simulator 📈 📉</h1>
<p> <b>Description:</b>
  a single player game.One of the most demanded skill in 2021 is stock prediction and analysis, this games aims in the similar aspect, you're given a couple of graphs and their respective standard deviations, you're task is to select the best performing stock. If you're prediction/analysis is true you will win else vice versa.<br><br>
NOTE: ! STILL UNDER CONSTRUCTION !
</p>

## Preview: ##
<p align="center">
<img src="preview img/graphs-1.png" height="700px" >
<img src="preview img/graphs-2.png" height="700px"  >
</p><br>
<hr style="border:2px solid gray"> </hr>


CHANGELOG.txt
Change Log
==========

1.0.0 (30/12/2021)
-------------------
- First Release


src/casino.py
import random
from time import sleep
def driverfunc(player):
    win_amou=0
    print("\n============================================================")
    print(numls)
    print("============================================================")
    chips=100
    print("---> Choose your desired number and place your bet <---")
    x1=int(input("\n>>>>>> Player-%s enter how many numbers u wanna place bet on: "%player))
    if x1>15:
        x1=input("choice must be within 15 nums, try again: ")
    while(x1>0):
        nlsv=int(input("enter number: "))
        mlsv=int(input("enter bet: $"))
        if (chips-mlsv)<0:
            print("insufficient balance!!, your left amount:",chips,",try again entering vaild amount..") 
            mlsv=int(input("enter bet: $"))
        else:
            nls.append(nlsv)
            mls.append(mlsv)
            chips-=mlsv
            print(f"-------------> remaining bal: ${chips}")
        x1-=1
    for i in nls:
        if i==win_num:
            index=int(nls.index(i))
            win_amou+=int(mls[index]*2)
            #print("chips earn: ",win_amou)
            mls.pop(index)
        else:
            continue
    #chips-=sum(mls)
    return (chips+win_amou),win_amou   
players=int(input("enter no.of player: "))
low=int(input("enter lower limit: "))
high=int(input("enter higher limit: "))
if high<15:
    high=int(input("higher limit must be greater than 15,enter higher limit: "))
nls=[]
mls=[]
wals=[]
for i in range(1,players+1):
    print(f'\nplayer-{i} you are allocated with $100')
numls=((random.sample(range(low,high),15)))
win_num=random.choice(numls)
print("cheat: ",win_num)
player_chips_ls=[]
for i in range(1,players+1):
    player_chips_ls_var,wa=driverfunc(i)
    player_chips_ls.append(player_chips_ls_var)
    wals.append(wa)
print("\n       $$$$ WINNINGS $$$$ \n")    
for i in range(0,players):
    sleep(0.5)
    print(f"=====> player-{i+1} you won: ${wals[i]} <==============")
    print(f"your total amount: ${player_chips_ls[i]}")
    if (player_chips_ls[i]-100)>0:
        print(f"profit: ${player_chips_ls[i]-100}\n")
    else:
        print(f"you lost: ${100-player_chips_ls[i]}\n")    
    


src/numguess.py
import random
players=int(input("enter no.of players: "))
c=1
print("-----------------------------------------")
print(".......Intiating Game for %s player......."%players)
print("-----------------------------------------")
low=int(input("enter lower limit: "))
high=int(input("enter higher limit: "))
numls=[]
rand=random.randint(low,high)
numls.append(rand)
guesslst=[]
user_input=int(input("guess a number between %s and %d:"%(low,high)))
while user_input!=rand:
    if c==(players*2):
        c=-1
        rand=random.randint(low,high)
        numls.append(rand)
        guesslst.append('new')
        print("---------Number changed------------")
    elif(user_input>rand):
        print("\nyour guess is just a bit higher")
        guesslst.append(user_input)
        user_input=int(input("try again: "))
    else:
        print("\noops! your guess is less than expected")
        guesslst.append(user_input)
        user_input=int(input("try again: "))
    c+=1    
print("\ncongratulations u guessed it right!!!")
if ((len(guesslst)+1)%players)==0:
    x=players
else:
    x=(len(guesslst)+1)%players
print("-----------------------------------------")
print("..............player %s won............."%(x))
print("-----------------------------------------")
print("\nit took",len(guesslst)+1,"guess to make a right choice")
print("\npreviously guessed numbers before guessing %i are:"%rand)
print("choosen numbers: ",numls)
print(guesslst)        

src/bin_kriket.py
#====================Version-1.5.0 score target working,added functions-improved logic================
import random
def bat_bowl(c,target,var_name):
    user_skore=0
    syst_skore=0
    hits=0
    print("\n------------------------------------------------------")
    print(f"   >>>>>>>>>>>>> ! {c+1}-innings !<<<<<<<<<<<")
    print("------------------------------------------------------\n")
    if c==1:
        print(f"------->Target: {target+1}<-------\n")
    user_chose=int(input("enter runs between 1-9: "))
    sys_chose=random.randint(1,9)
    print(f"                              {sys_chose}: computer choice <<<<<<")
    user_skore+=user_chose
    syst_skore+=sys_chose
    if c==1 and var_name=="you're":
        hits=user_skore
    elif c==1 and var_name=="computer":
        hits=syst_skore
    if sys_chose==user_chose:
        user_skore-=user_chose
        syst_skore-=sys_chose
        print("\n=====================================")
        print(f"-----------{var_name} duck out-------------")
        print("======================================")
    else:
        while(sys_chose!=user_chose and hits<=target+1):
            user_chose=int(input(("enter runs between 1-9: ")))
            sys_chose=random.randint(1,9)
            if int(user_chose)>9:
                user_chose=input(("error!, enter runs between 1-9: "))
            if sys_chose!=user_chose:
                print(f"                              {sys_chose}: computer choice <<<<<<")
                user_skore+=user_chose
                syst_skore+=sys_chose
            else:
                print(f"                              {sys_chose}: computer choice <<<<<<")
                print("\n=====================================")
                print(f"-----------{var_name} out-------------")
                print("======================================")
            if c==1 and var_name=="you're":
                hits=user_skore
            elif c==1 and var_name=="computer":
                hits=syst_skore
    if var_name=="computer":
        print(f"\n----->{var_name} score is : {syst_skore}<-----")
    else:
         print(f"\n----->{var_name} score is : {user_skore}<-----")
    return user_skore,syst_skore
def stats(player1,player2):
    if player1>player2:
        print("\n============== ! stats ! ============\n")
        print(f"$ you're score is {player1}")
        print(f"$ computer score is {player2}")
        print(f"$ you won the match by {player1-player2} runs")
    else:
        print(f"$ you're score is {player1}")
        print(f"$ computer score is {player2}")
        print(f"$ computer won the match by {player2-player1} runs")
def toss_win(player_choice):
    c=0
    if int(player_choice)==0:
        player1,t=bat_bowl(c,999,"you're")
        c+=1
        t,player2=bat_bowl(c,player1,"computer")
        stats(player1,player2)
    else:
        t,player2=bat_bowl(c,999,"computer")
        c+=1
        player1,t=bat_bowl(c,player2,"you're")
        stats(player1,player2)
print("======================================================")
print("--welcome to hand kriket or more like binary kriket--")
print("======================================================\n")
ls=['b','bw']
toss=input(("choose toss--> enter 'h' for heads or 't' tails : "))
win_toss=random.choice(['h','t'])
if win_toss==toss:
    player_choice=input("you won the toss--> enter 'b' for bat or 'bw' for bowl: " )
    toss_win(ls.index(player_choice))
else:
    player_choice=random.choice([0,1])
    if player_choice==0:
        print(f"computer won the toss--> computer chooses to bat") 
    else:
        print(f"computer won the toss--> computer chooses to bowl") 
    toss_win(~player_choice+2)




#===================<PREVIOUS VERSIONS>================

#====================Version-1.0.0 improved--score,toss working================
# import random
# print("======================================================")
# print("--welcome to hand kriket or more like binary kriket--")
# print("======================================================\n")
# c="bowl"
# def bat_bowl(var_name,ini_num,target):
#     print("\n------------------------------------------------------")
#     print(f"   >>>>>>>>>>>>> ! {ini_num} innings !<<<<<<<<<<<")
#     print("------------------------------------------------------\n")
#     if ini_num=="second":
#         print(f"------->Target: {target+1}<-------\n")
#     sys_chose=0
#     user_skore=0
#     syst_skore=0
#     user_chose=int(input("enter runs between 1-9: "))
#     sys_chose=random.randint(1,9)
#     print(f"                              {sys_chose}: computer choice <<<<<<")
#     user_skore+=user_chose
#     syst_skore+=sys_chose 
#     if sys_chose==user_chose:
#         user_skore-=user_chose
#         syst_skore-=sys_chose
#         print(f"{var_name} duck out")
#     else:
#         while((sys_chose!=user_chose) or (user_skore<=int(target)) or (syst_skore<=int(target))):
#             user_chose=int(input(("enter runs between 1-9: ")))
#             sys_chose=random.randint(1,9)
#             if int(user_chose)>9:
#                 user_chose=input(("error!, enter runs between 1-9: "))
#             if sys_chose!=user_chose:
#                 print(f"                              {sys_chose}: computer choice <<<<<<")
#                 user_skore+=user_chose
#                 syst_skore+=sys_chose
#             else:
#                 print(f"                              {sys_chose}: computer choice <<<<<<")
#                 print("\n=====================================")
#                 print(f"-----------{var_name} out-------------")
#                 print("======================================")
#             #user_skore+=user_chose
#             #syst_skore+=sys_chose
#     if var_name=="computer":
#         print(f"\n----->{var_name} score is : {syst_skore}<-----")
#     else:
#          print(f"\n----->{var_name} score is : {user_skore}<-----")
#     return user_skore,syst_skore
# toss=input(("choose toss--> 'h' for head or 't' tails : "))
# win_toss=random.randint(1,2)
# if (win_toss==1 and toss=='h') or (win_toss==2 and toss=='t'):
#     user_choice=input("you won the toss--> 'b' for bat or 'bw' for bowl: " )
#     if user_choice=="b":
#         u_skore,t=bat_bowl("you're","first",9999)
#         t,sys_skore=bat_bowl("computer","second",u_skore)
#     else:
#         t,sys_skore=bat_bowl("computer","first",9999)
#         u_skore,t=bat_bowl("you're","second",sys_skore)
# else:
#     sys_choice=random.randint(1,2)
#     if sys_choice==1:
#         c="bat"
#         print(f"computer won the toss--> computer chooses to {c}")
#         t,sys_skore=bat_bowl("computer","first",9999)
#         u_skore,t=bat_bowl("you're","second",sys_skore)
#     else:
#         print(f"computer won the toss--> computer chooses to {c}")
#         u_skore,t=bat_bowl("you're","first",9999)
#         t,sys_skore=bat_bowl("computer","second",u_skore)
# if u_skore>sys_skore:
#     print("\n============== ! stats ! ============\n")
#     print(f"$ you're score is {u_skore}")
#     print(f"$ computer score is {sys_skore}")
#     print(f"$ you won the match by {u_skore-sys_skore} runs")
# else:
#     print(f"$ you're score is {u_skore}")
#     print(f"$ computer score is {sys_skore}")
#     print(f"$ computer won the match by {sys_skore-u_skore} runs")


src/lottery_simulator.py
import random
print("[[[[[[[[[[[[[[[]]]]]]]]]]]]]]]")
print("        TEST YOUR LUCK!!!      ")
print("[[[[[[[[[[[[[[[]]]]]]]]]]]]]]]\n")
players=int(input("enter how many player: "))
print("-----------------------------------------")
print("........Game intiating for %s player......"%players)
print("-----------------------------------------")
player1=[]
player2=[]
player3=[]
player4=[]
x1=0
x2=0
x3=0
x4=0
low=random.randint(0,1000)
high2=random.randint(low,1000000)
high=random.randint(low,high2)
lucky_num=random.randint(low,high)
count=0
while lucky_num not in (player1 or player2 or player3 or player4):
    for i in range(1,players*10+1):
        if i<11 and x1<1:
            player1.append(random.randint(low,high))
        if (i>10 and i<21) and x2<1:
            player2.append(random.randint(low,high))
        if (i>20 and i<31) and x3<1:
            player3.append(random.randint(low,high))
        elif i>40 and x4<1:
            player4.append(random.randint(low,high))
    if lucky_num in player1:
        x1+=1
    else:
        player1=[]
    if lucky_num in player2:
        x2+=1
    else:
        player2=[]
    if lucky_num in player3:
        x3+=1
    else:
        player3=[]
    if lucky_num in player4:
        x4+=1
    else:
        player4=[]
    count+=1
print("\n=========================================")
print(f"----! lucky number is: {lucky_num} !----")
print("===========================================")
if lucky_num in player1:
    print("---------------------------------------------------------------------------------")
    print(">>>> Congratulations player 1, you won points %s"%(count+player1.index(lucky_num)))
    print("-------------------------------------------------------------------------------\n")
    print("$--> your list:",player1)
elif lucky_num in player2:
    print("---------------------------------------------------------------------------------")
    print(">>>> Congratulations player 2, you won points %s"%(count+player2.index(lucky_num)))
    print("-------------------------------------------------------------------------------\n")
    print("$--> your list:",player2)
elif lucky_num in player3:
    print("---------------------------------------------------------------------------------")
    print(">>>> Congratulations player 3, you won points %s"%(count+player3.index(lucky_num)))
    print("-------------------------------------------------------------------------------\n")
    print("$--> your list:",player3)
elif lucky_num in player4:
    print("---------------------------------------------------------------------------------")
    print(">>>> Congratulations player 4, you won points %s"%(count+player4.index(lucky_num)))
    print("-------------------------------------------------------------------------------\n")
    print("$--> your list:",player4)
print("\n==========TERIMINATING===============")

src/graphs.py
# import matplotlib.pyplot as plt
# import random
# import statistics
# i=10
# while(i>0):
#     sdl=[]
#     x1 = ((random.sample(range(0,100),30)))
#     x1.sort()
#     x2 = ((random.sample(range(100,150),30)))
#     x2.sort()
#     x1.extend(x2)
#     def yplot(y1,sdl):
#         for i in range(11):
#             var_low=random.randint(0,100)
#             var_high=random.randint(var_low,100)
#             if (var_high-var_high)<5:
#                 var_high+=5
#             y_var = ((random.sample(range(var_low,var_high),5)))
#             y_var.sort()
#             a=min(y_var)
#             b=max(y_var)
#             sdl.append(a)
#             sdl.append(b)
#             y1+=y_var
#         return y1
#     y1 = ((random.sample(range(0,100),5)))
#     y1.sort()
#     a=min(y1)
#     b=max(y1)
#     sdl.append(a)
#     sdl.append(b)
#     yplot(y1,sdl)
#     print(f"{i}) Standard Deviation of sample is %s"% (statistics.stdev(sdl)))
#     plt.plot(x1, y1)
#     plt.xlabel('x - axis')
#     plt.ylabel('y - axis')
#     plt.title('Your stock performance')
#     plt.show()
#     i-=1


import random
print("======================================================")
print("--welcome to hand kriket or more like binary kriket--")
print("======================================================\n")
c="bowl"
def bat_bowl(var_name,ini_num,target):
    print(f"\n>>>>>>>>>>>>> ! {ini_num} innings !<<<<<<<<<<<\n")
    if ini_num=="second":
        print(f"------->Target: {target+1}<-------\n")
    sys_chose=0
    user_skore=0
    syst_skore=0
    user_chose=int(input("enter runs between 1-9: "))
    sys_chose=random.randint(1,9)
    print(f"                              {sys_chose}: computer choice <<<<<<")
    if sys_chose==user_chose:
        print(f"{var_name} duck out")
    else:
        while((sys_chose!=user_chose) or (user_skore>=int(target)) or (syst_skore>=int(target))):
            user_chose=int(input(("enter runs between 1-9: ")))
            if int(user_chose)>9:
                user_chose=input(("error!, enter runs between 1-9: "))
            sys_chose=random.randint(1,9)
            if sys_chose!=user_chose:
                print(f"                              {sys_chose}: computer choice <<<<<<")
                user_skore+=user_chose
                syst_skore+=sys_chose
            else:
                print(f"                              {sys_chose}: computer choice <<<<<<")
                print("\n=====================================")
                print(f"-----------{var_name} out-------------")
                print("======================================")
            user_skore+=user_chose
            syst_skore+=sys_chose
    if var_name=="computer":
        print(f"\n----->{var_name} score is : {sys_skore}<-----")
    else:
         print(f"\n----->{var_name} score is : {sys_skore}<-----")
    return user_skore,syst_skore
toss=input(("choose toss--> 'h' for head or 't' tails : "))
win_toss=random.randint(1,2)
if (win_toss==1 and toss=='h') or (win_toss==2 and toss=='t'):
    user_choice=input("you won the toss--> 'b' for bat or 'bw' for bowl: " )
    if user_choice=="b":
        u_skore,sys_skore=bat_bowl("you're","first",9999)
        sys_skore,u_skore=bat_bowl("computer","second",u_skore)
    else:
        sys_skore,u_skore=bat_bowl("computer","first",9999)
        u_skore,sys_skore=bat_bowl("you're","second",sys_skore)
else:
    sys_choice=random.randint(1,2)
    if sys_choice==1:
        c="bat"
        print(f"computer won the toss--> computer chooses to {c}")
        sys_skore,u_skore=bat_bowl("computer","first",9999)
        u_skore,sys_skore=bat_bowl("you're","second",sys_skore)
    else:
        print(f"computer won the toss--> computer chooses to {c}")
        u_skore,sys_skore=bat_bowl("you're","first",9999)
        sys_skore,u_skore=bat_bowl("computer","second",u_skore)
if u_skore>sys_skore:
    print("============== ! stats ! ============")
    print(f"$ you're score is {u_skore}")
    print(f"$ computer score is {sys_skore}")
    print(f"$ you won the match by {u_skore-sys_skore} runs")
else:
    print(f"$ you're score is {u_skore}")
    print(f"$ computer score is {sys_skore}")
    print(f"$ computer won the match by {sys_skore-u_skore} runs")







